ipld/plugin/nmt.go

```diff
@@ -53,9 +53,24 @@ func NewNamespaceHasher(hasher *nmt.Hasher) hash.Hash {
 }
 
 func (n *namespaceHasher) Write(data []byte) (int, error) {
-	n.tp = data[0]
+	ln, nln, hln := len(data), int(n.NamespaceLen), n.Hash.Size()
+	switch ln {
+	default:
+		return 0, fmt.Errorf("wrong data size")
+	// inner node
+	case (nln*2+hln)*2 + 1: // with type byte
+		fallthrough
+	case (nln*2 + hln) * 2:
+		n.tp = nmt.NodePrefix
+	// leaf data node
+	case nln + 256:
```

Comment:
Please use the constant from core instead of `256`. Also, note that we will change to 512 byte shares soon: https://github.com/celestiaorg/celestia-core/issues/724

Reply:
Thanks for noting that, @liamsi.  I thought about that, but then the cycle dependency issue stopped me from doing that(ipld/plungin cannot import ipld)but this anyway has to be fixed in this PR. The dependency cycle will be resolved by #878, but before that I will just import directly from App.

Reply:
Done